docs/architecture/sdk/index.md

```diff
@@ -9,38 +9,77 @@ SDK for the Bitwarden [Password Manager][pm]. The SDK is written in Rust and pro
 multiple languages. The general end goal of the SDK is to own everything up to the presentational
 layers. This includes but is not limited to: API calls, data models, encryption, and business logic.
 
-<Bitwarden>We have collected a list of various learning resources on
-[Confluence](https://bitwarden.atlassian.net/wiki/spaces/DEV/pages/517898288/Rust+Learning+Resources).</Bitwarden>
+<Bitwarden>We have compiled a list of resources for learning Rust in a
+[Confluence page](https://bitwarden.atlassian.net/wiki/spaces/DEV/pages/517898288/Rust+Learning+Resources).</Bitwarden>
+For API documentation view the latest
+[API documentation](https://sdk-api-docs.bitwarden.com/bitwarden/index.html) that also includes
+internal private items.
 
 ## Architecture
 
 The Bitwarden SDK is structured as a single [Git repository](https://github.com/bitwarden/sdk) with
-multiple internal crates. Review the repository README for more information about the different
-crates.
+multiple internal crates. Please review the `README` in the repository for up to date information
+about the different crates.
 
-We generally strive towards extracting features into separate crates to keep the core `bitwarden`
+We generally strive towards extracting features into separate crates to keep the `bitwarden-core`
 crate as lean as possible. This has multiple benefits such as faster compile-time and clear
 ownership of features.
 
 ### `bitwarden` crate
 
-The `bitwarden` crate is the main entry point for the SDK and is responsible for wiring up the sub
-crates. It contains a `Client` struct which represents a single account instance in the SDK. The
-`Client` struct implements multiple callable methods that may manipulate the state of the SDK.
+The `bitwarden` crate represents the entry point for consumers of the SDK and is responsible for
+providing a cohesive interface. The crate re-exports functionality of the internal crates and
+contains very little logic itself.
+
+### `bitwarden-core` crate
+
+The `bitwarden-core` crate contains the underlying functionality of the SDK. This includes a
+`Client` struct. Other crates in the SDK depend on `bitwarden-core` and provide extensions to the
+`Client` struct to implement specific domains.
+
+## Client struct
+
+The `Client` struct is the main entry point for the SDK and represents a single account instance.
+Any action that needs to be performed on the account is generally done through the `Client` struct.
+This allows the internal to be hidden from the consumer and provides a clear API.
+
+We can extend the `Client` struct using extension traits in feature crates. This allow the
+underlying implementation to be internal to the crate with only the public API exposed through the
+`Client` struct. Below is an example of a generator extension for the `Client` struct.
+
+```rust
+pub struct ClientGenerator<'a> {
```

Comment:
**thought**: I know this is an established format but to me `ClientGenerator` sounds weird, like it's something that generates clients. It's the same thing with the other client extensions, e.g. `ClientVault` which sounds like there are multiple vaults and this one belongs to the `Client`.

On the other hand `ClientGeneratorExt` sounds very reasonable, it's a `Generator Extension` for the `Client`.

I don't think I would've reacted to this if we would've just passed around the `Client` struct and simply applied `Extensions` to it, i.e. `Client.password(...)`, but as it is now, we are actually creating secondary "sub-clients` which themself have a reference to the `Client` instance. `Client.generator()` actively constructs a new type of "Client"

Maybe I'm just nitpicking I don't know, but I think we'll have problems referring to things in the future

Reply:
I'm open for alternative name schemas.

Reply:
Well my suggestion is to flip the name from `ClientGenerator` to `GeneratorClient`, not that big change considering how big my previous message was xD

Reply:
@Hinton 
Well my suggestion is to flip the name from ClientGenerator to GeneratorClient, not that big change considering how big my previous message was xD

Reply:
Seems fine to me. Mind creating a jira issue and tackling this mountain of a task :).

Reply:
@Hinton Done! [PM-13166 Rename SDK sub-clients to `<>Client` instead of `Client<>`](https://bitwarden.atlassian.net/browse/PM-13166)